app/src/main/java/com/cmput301f20t21/bookfriends/ui/library/OwnedViewModel.java

```diff
@@ -85,17 +83,12 @@ private void fetchBooks() {
                                 return;
                             }
                             List<DocumentSnapshot> documents = result.getDocuments();
-                            bookData.addAll(IntStream.range(0, documents.size()).mapToObj(i -> {
-                                DocumentSnapshot document = documents.get(i);
-                                Book book = bookRepository.getBookFromDocument(document);
-                                if (document.get("imageName") != null) {
-                                    bookRepository.getImage((String) document.get("imageName")).addOnSuccessListener(uri -> {
-                                        book.setImageUri(uri);
-                                        updatedPosition.setValue(i);
-                                    });
-                                }
-                                return book;
-                            }).collect(Collectors.toList()));
+                            bookData.clear();
```

Comment:
not sure why, unlike the borrowed vm where I set array directly on the live data, if I do it here (like directly `books.setValue(...stream...collect...)`) the list is empty. Didn't bother to look into it.